### DISCUSSION TITLE
Teams and Projects

### DISCUSSION THREAD

**Discussion by erlend-sh** [muni-town/roomy#324] [Ideas]
To effectively build ambitious projects you need teams, and teams are formed around projects. They are two sides of the same coin.

*note:* Some of this was written while Roomy (formerly Commune) and Weird (personal CMS) were two separate apps. They’ve since merged into one.

- - -

One thing current collaboration apps (inspired by “agile” and other such rigid/busy practices) generally do poorly is to help people work on their projects in perpetuity. In mainstream productivity frameworks, leaving a task "hanging" is considered bad practice.

Depending on how you define a project I've been involved with 20-200 of them. The thing is, I’ve never fully given up on any of a single one. But I’m detached enough (because I had to narrow my focus to get anything of substance done) to happily play the support-role on most of them. I just wanna see all those project ideas be brought into being, whichever way possible.

Having spent some 20 years now doing projects (by building/joining/assisting teams), I see that ideas from decades ago aren’t standing still, even if they’ve just had a few days worth of active development time. And then there are global-scale dreams like “let’s make an open source game console & marketplace owned by the commons” which we are in fact becoming rapidly close to (as long as venture capital and AI don't enshittify it first).

Point being, there’s basically no downside and loads of upside to listing every single project-idea I’ve ever engaged with. Even though the chances are small, just having any chance of making contact with someone whose idea is closely aligned enough with mine that we can team up? Pure upside ✨

Weird (identity) concerns itself with the listing of projects & teams; directories for discovery.
Roomy (bonfire) concerns itself with the communications of projects & teams; spaces for gathering.

I think the act of making the right OAuth scopes on either side will provide an interesting exercise that might point in the direction of additional spec work we might need to make up for our own ends. Where Commune and Weird meet, we are making a Protocol for Collaboration 🫂 

Inspiration: https://portfolo.app/
Research *[projects](https://github.com/okhat/blog/blob/main/2024.09.impact.md)*

**Comment by erlend-sh** [comment]
Regarding MVP of the [network of shared purpose](https://github.com/commune-os/weird/issues/3): Another way of putting it is that every project profile will have a /team page. That page will be much like any company’s standard team page. But it can scale from 3x3 roomy grid to compact 16x100+

Probably all personal profiles should have a /team page as well. All of my project collaborators are basically on 'Team Erlend', and I wanna list them as such. My team page would be made up of all of my public contacts, ordered roughly by the degree of entanglement (time spent working together) between me and each person listed.

Team pages would have some small randomizations built in. It’s not supposed to be a page that encourages playing the comparison-game; if it devolves into a numbers game then it’s a boring/bad game.

My personal team page would be made up of my connections on:
- GitHub
- discord
- email 
- ActivityPub
- and personal websites, for those who have one.

It’s like the opposite of this: https://indieweb.org/POSSE
POSSE is the ideal end-game, but it [doesn’t work as a starting point](https://darthmall.net/weblog/2024/indieweb-is-for-devs/) for regular folks.

Any number of ‘activity logs’ (aka ‘recorded proofs of work’) can be plugged into the teams. A good start would be GitHub and Discord. What we are looking for is basic activity data that signifies project involvement. On GitHub that’s recent commits/comments. On Discord that’s recent posts.

**Comment by erlend-sh** [comment]
Musing about teams/projects: I think every project has at least one team, I.e. the project-wide team of members with write-access to that project.

I don’t think we should allow creating stand-alone teams, at least not until we have a very clear use case for them. So to start with, every _team_ needs to exist within a project. A team without a project is just a group dm.

We could probably make a v0.1 of Project(+Team) by duplicating our existing Links component* (*no standardized language here yet). It's just a Links page with a member-list attached.

The start of Weird’s own project page would just be a series of links to essential resources, like this Discord (and hopefully soon a public-web archive of it), our github, blog posts…

v0.1 of a Project is literally just a linkspage (which for starters can be single-user-owned, even though org-owned will be the default later) with a subpage that lists a subset of our network’s member list.

That’s kind of our page-building philosophy: just start piling some stuff up. If enough people pile on (good-heartedly), it’ll take on a clearer shape.

**Reply by erlend-sh** [reply]
### Mockup - Project Nodes
![project-weird-mockup](https://github.com/user-attachments/assets/d56cd0a1-b58e-48d2-ac31-9bd7634ec9b6)

Forgot to add a top-level link for *Members*

*edit:* would call them nodes/extensions.. instead of *spaces*, which is an occupied term.

**Reply by erlend-sh** [reply]
https://news.ycombinator.com/item?id=42757462

> I still feel GH Projects is solely aimed at OSS or dev-centric or dev-only orgs and doesn't cater to teams with non-devs, which is how I think most orgs function. I'm not sure if it'll ever try to be something more than that but I really wish it did.

**Comment by erlend-sh** [comment]
We’ve been increasingly referring to the common layer of Teams/Projects underpinning both Weird & Roomy as *Spaces*.

We’ve talked before about how Roomy as a product would probably be an extension of the weird.one platform (or the other way around), which I think still holds.

My current thinking: Roomy-as-a-Service will be sold as ~~Weird Spaces~~ Roomy Spaces at `roomy.space`

Weird Spaces = [Chatty community gardens](https://blog.muni.town/chatty-community-gardens/) = Roomy *[spaces](https://github.com/muni-town/roomy/discussions/21)*

So, (Roomy) Spaces-as-a-Service, the 🆕 SaaS.

It would cost $50/month and upwards for Business Customers (80% off for big non-profits, free for all small non-commercial groups).

We would also continue to build out our free-for-all (individuals, not corporations) p2p offering co-existing with (and adding robustness to) that service.

![notionhome_td8vd5_shrink_c6pnkg](https://github.com/user-attachments/assets/a61463f4-c44b-4824-bc7d-43678d41a15b)

☝️  group-projects, as all the most ambitious projects are, aren’t coordinated primarily in the *planning* layer (Notion/Trello, GitHub Issues..). 90% of distributed project coordination happens in the group chat, at the very least in terms of ‘messages sent’.

..that holds true for all the projects *I enjoy* anyway 😄 I want as little uppercase Project Management to be going on as at all possible.

That’s one of the reasons why Notion didn’t stick with me even though it was an excellent tool for Project-Drafting. It just wasn’t essential enough in the day-to-day of projects **work**.

Roomy is an all-in-one toolkit for *public-first project *spaces**. Projects very much in *plural*, not existing as isolated islands.

(As a private-first network for activist communities, Acter provides some excellent prior art to concretize what constitutes a [Space](https://acter.global/guide/space-user-guide/).)

> How is a Roomy space an ‘all-in-one projects space’?

The same way Discord is 90% of our projects space right now. 90% of our most collaborative “project-ing time” is spent communicating together in our shared group-comms space (currently Discord, very soon Roomy).

The Roomy app experience is this: <https://blog.muni.town/chatty-community-gardens/>

Basically Discord meets Notion. As two different app-windows, Roomy is Discord-like first while Weird is Notion-like first, but they’re both views into the same underlying core for a ‘Digital Comms App’ (in-a-box). Similarly, they’re both facilitators of *space-making*.

The notion-first side of the experience is demonstrated very effectively here: https://maggieappleton.com/nontechnical-gardening

**Comment by erlend-sh** [comment]
The Circles community currency has been on our radar for a bit now:

https://aboutcircles.com/
https://aboutcircles.com/manifesto
https://www.gnosis.io/blog/building-the-open-internet
https://metri.xyz/news/circles-and-crc-fundamentals
https://hackernoon.com/circles-a-global-experiment-to-reinvent-money
https://youtu.be/x_RUVF2DQlk?si=ATgOIL4oqT8HQEs2

https://docs.aboutcircles.com/

> Circles is a decentralized protocol designed to create and distribute fair and social money through personal currencies. At the core of the Circles project lies a simple yet powerful principle:
> 
> "Every person receives one Circle every hour, unconditionally."
Circles uses smart contracts, deployed on Gnosis Chain, for creation of personal and group Avatars and their currencies (CRC). Circles as a network works via trust connection.
> 
> To join Circles 2.0, you would require an invitation from a human/personal avatar. The pathfinder service is deployed to find the optimal path between trust connections to exchange personal and group currencies. Circles smart contracts are based on ERC1155 multi-token standard for both personal avatars and group avatars.
> 
> Every person receives 1 CRC every hour, hence 24 CRC per single day.
Circles undergo daily demurrage at a rate equivalent to 7% per year. Issuance for past days accounts for this demurrage, ensuring fair distribution over time.

And just like Roomy it makes Groups a foundational building block: https://aboutcircles.com/groups

This could be a useful foundation for [local community currencies](https://grassrootseconomics.org).
